scenario_player/utils/token.py

```diff
@@ -409,6 +414,6 @@ def deposit(self, target_address) -> Union[str, None]:
             return
 
         log.debug("deposit call required - insufficient funds")
-        deposit_amount = max_funding - balance
+        deposit_amount = total_deposit + max(max_funding, min_deposit) - balance
```

Comment:
`max()` should not be necessary here - there should be a check<sup>[suggestion](#myfootnote1)</sup>

 in [this class](https://github.com/raiden-network/scenario-player/blob/master/scenario_player/utils/configuration/settings.py#L40:L61) instead, that ensures that `max_funding >= balance_per_node` when loading the scenario yaml.

Otherwise, this would silently use a value that may not be expected by the user.

-----------------------------------------------------------------
<a name="myfootnote1">Suggested validation:</a>:

```
def validate(self):
    self.assert_option(self.max_funding < self.balance_per_node, "max_funding must be >= balance_per_node value!")
```

I trust you can write a test for that just fine. ;P